libs/query-state/src/lib/operators/tap-state.ts

```diff
@@ -24,6 +33,11 @@ export function tapState<QueryData>(callbacks: {
 
       case 'revalidate':
         callbacks.onRevalidate?.();
+
+        if (options?.optimisticUpdates) {
```

Comment:
I'm probably missing something but how does this "enable" optimistic updates?

Reply:
Here's a recording of what the desired result is. In the data.service I added an updated field to illustrate that once the api call completes, you will see the real updates.

```
 delay(1000),
      map((person) => {
        person.updatedAt = Date.now().toString();
        return person;
      })
```

![Kapture 2022-05-08 at 17 49 44](https://user-images.githubusercontent.com/10077295/167317253-f8a1ae1d-5750-437e-85ca-96ec02cbdc83.gif)

Reply:
Oh I see, I don't know if `optimisticUpdates` is the correct name then.
What about `preload`?
Could you also add a test for this please.

Reply:
I’m ok with renaming the setting! Preload sounds a lot like prefetch. But I do like your first suggestion, refresh. Maybe backgroundRefresh?

Should the setting be able to be globally set to a default?

I will add tests and update with new examples once things are decided

Reply:
Sorry, after giving it some more thought I think we can make this possible without an additional config flag.
The data should be passed to the revalidate method of `tapState`.
The component code then looks like this if you want to use this future.
Notice the extra if statement because `person` will be undefined the first time because it isn't in the cache.

```diff
  ngOnInit(): void {
    this.queryState.effect(
      this.queryState.data$.pipe(
        tapState({
          onSuccess: (person) => {
            this.model.id = person.id;
            this.model.firstname = person.firstname;
            this.model.lastname = person.lastname;
          },
+          onRevalidate: (person) => {
+            if (person) {
+              this.model.id = person.id;
+              this.model.firstname = person.firstname;
+              this.model.lastname = person.lastname;
+            }
          },
        })
      )
    );
  }
```

Reply:
I mentioned that solution initially on our discord conversation, but I think that the downside of this is that you will have to wire up all your components on both onSuccess, onRevalidate and onError, etc.

What I was thinking is that we can with some configuration set some of these things up so that we get the desired output with less boiler plate need. Like shown in this PR.

Anyone who wants custom handlers for each tap, can have that as well.

but these are just thoughts. I don’t think strongly about one over the other, so I’ll leave it up to you to decide :)